gui/default/syncthing/core/syncthingController.js

```diff
@@ -521,6 +527,19 @@ angular.module('syncthing.core')
             console.log("recalcCompletion", device, $scope.completion[device]);
         }
 
+        function replaceAddressPort(address, newPort) {
+            var lastColonIndex = address.length;
+            for (var index = 0; index < address.length; index++) {
+                if (address[index] === ":") {
+                    var lastColonIndex = index;
```

Comment:
Is the var in a loop body required? I'm no JS pro, but I fear that it's somehow scoped within the if statement, and reverts to it's old value outside of the if/for?

Reply:
I will change it to let.

Reply:
everything should probably be let until var is needed. I will change.

Reply:
I do not think that using `let` is a good idea due to it being unsupported in older browsers. Syncthing right now uses only `var` in all its JS files.

Reply:
I agree that the double `var` declaration looks fishy, as if a different variable is created within the loop as opposed to updating the existing one.

Reply:
Please remove the `var` in front of `lastColonIndex`. It is already initialized before.